src/Chapter4.hs

```diff
@@ -307,6 +310,11 @@ data List a
     = Empty
     | Cons a (List a)
 
+instance Functor List where
+  fmap :: (a -> b) -> List a -> List b
+  fmap _ Empty = Empty
+  fmap f (Cons head tail) = Cons (f head) (fmap f tail)
```

Comment:
`head` and `tail` are the names of standard functions. Haskell allows name shadowing, that's why it works. Though, in practice, it's usually better to avoid name clashes with other functions cause it could be confusing 🙂

Reply:
Oh, nice you noticed that! In Elixir, the functions are `hd` and `tl`, so I got used to write `head` and `tail` for the variables :sweat_smile:

Reply:
Yeah, for Haskell standard naming conventions for variables are usually `x` and `xs`. It may look not-so-readable in the beginning, but after some time you get used to it. Most of the time names don't matter, because functions are polymorphic enough and there's only one reasonable implementation for such polymorphic functions :slightly_smiling_face: